types/module/module.go

```diff
@@ -328,6 +341,30 @@ func (m *Manager) ExportGenesis(ctx sdk.Context, cdc codec.JSONMarshaler) map[st
 	return genesisData
 }
 
+// MigrationHandler is the migration function that each module registers.
+type MigrationHandler func(store sdk.Context, storeKey sdk.StoreKey, cdc codec.Marshaler) error
```

Comment:
```suggestion
type MigrationHandler func(store sdk.Context) error
```

All this stuff is not needed. Let's keep the AP simple